tests/ui/traits/const-traits/call-const-closure.rs

```diff
@@ -0,0 +1,22 @@
+//@ compile-flags: -Znext-solver
+//@ edition:2021
+
+#![feature(const_trait_impl, effects, const_closures)]
+#![allow(incomplete_features)]
+
+#[const_trait]
+trait Bar {
+    fn foo(&self);
+}
+
+impl Bar for () {
+    fn foo(&self) {}
+}
+
+const FOO: () = {
+    (const || ().foo())();
+    //~^ ERROR the trait bound `(): ~const Bar` is not satisfied
```

Comment:
This is the other known bug `tests/crashes/118320.rs`. The error message is wrong, since we should not be enforcing `~const Bar` but `const Bar` here, probably. Const closures need a total rework and redesign, though, so I'm gonna leave it.